lxd/main_callhook.go

```diff
@@ -24,14 +32,175 @@ func (c *cmdCallhook) Command() *cobra.Command {
   Call container lifecycle hook in LXD
 
   This internal command notifies LXD about a container lifecycle event
-  (start, stopns, stop, restart) and blocks until LXD has processed it.
+  (start, startmountns, stopns, stop, restart) and blocks until LXD has processed it.
 `
 	cmd.RunE = c.Run
 	cmd.Hidden = true
 
+	// devicesRootFolder is used to specify where to look for CDI config device files.
+	cmd.Flags().StringVar(&c.devicesRootFolder, "devicesRootFolder", "", "Root folder for CDI devices")
+
 	return cmd
 }
 
+// resolveTargetRelativeToLink converts a target relative to a link path into an absolute path.
+func resolveTargetRelativeToLink(link string, target string) (string, error) {
+	if !filepath.IsAbs(link) {
+		return "", fmt.Errorf("The link must be an absolute path: %q (target: %q)", link, target)
+	}
+
+	if filepath.IsAbs(target) {
+		return target, nil
+	}
+
+	linkDir := filepath.Dir(link)
+	absTarget := filepath.Join(linkDir, target)
+	cleanPath := filepath.Clean(absTarget)
+	absPath, err := filepath.Abs(cleanPath)
+	if err != nil {
+		return "", err
+	}
+
+	return absPath, nil
+}
+
+// customCDILinkerConfFile is the name of the linker conf file we will write to
+// inside the container. The `00-lxdcdi` prefix is chosen to ensure that these libraries have
+// a higher precedence than other libraries on the system.
```

Comment:
Why do they need a higher precedence, please can you add to the comment.

Is this what the old method used?

Reply:
Is this so they get logged in the liblxc log file?

Reply:
This is a pattern that is used in `nvidia-ctk cdi hook` (see this: https://github.com/NVIDIA/nvidia-container-toolkit/blob/b7948428ffc5dff7bdd05aa4fb1f77d3c333ab2d/cmd/nvidia-cdi-hook/update-ldcache/update-ldcache.go#L156C1-L161C69)

Reply:
very good, ta